lib/database.js

```diff
@@ -150,14 +150,15 @@ module.exports = class Database extends EventEmitter {
 
         return this.connection;
       });
+    } else {
+      return Promise.reject(new Error(`Connection string of type ${typeof this.connectionString} not supported.`));
```

Comment:
Previously, `connect` returned `undefined` if the connection string was an supported type. Now, we'll reject the promise instead to match the format of the rest of the function.

Reply:
`s/supported/unsupported`?